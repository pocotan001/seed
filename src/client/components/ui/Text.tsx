import styled from "styled-components";
import color, { ColorProps } from "../styles/extends/color";
import margin, { MarginProps } from "../styles/extends/margin";
import typography, { TypographyProps } from "../styles/extends/typography";

type TextProps = TypographyProps & ColorProps & MarginProps;

const Text = styled("span")<TextProps>`
  ${typography};
  ${color};
  ${margin};
`;

export default Text;
