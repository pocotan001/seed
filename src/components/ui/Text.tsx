import styled from "styled-components";
import withColor, { ColorProps } from "~/components/styles/extends/withColor";
import withFont, { FontProps } from "~/components/styles/extends/withFont";
import withMargin, {
  MarginProps
} from "~/components/styles/extends/withMargin";

interface TextProps extends ColorProps, FontProps, MarginProps {
  children: React.ReactNode;
}

const Text = styled<TextProps, "span">("span")`
  ${withFont};
  ${withColor};
  ${withMargin};
`;

export default Text;
