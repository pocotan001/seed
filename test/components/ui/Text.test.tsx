import * as React from "react";
import * as renderer from "react-test-renderer";
import Text from "~/components/ui/Text";

describe("<Text>", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<Text>alo</Text>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
