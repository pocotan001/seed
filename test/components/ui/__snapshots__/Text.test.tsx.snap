// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text> should render correctly 1`] = `
<span
  className="sc-bdVaJa hXSkgu"
>
  alo
</span>
`;
